itou/emails/tasks.py

```diff
@@ -0,0 +1,171 @@
+
+    Returns a **list** of "sanitized" emails.
+    """
+
+    if len(email_message.to) <= _MAILJET_MAX_RECIPIENTS:
+        # We're ok, return a list containing the original message
+        return [email_message]
+
+    sanitized_emails = []
+    to_chunks = chunks(email_message.to, _MAILJET_MAX_RECIPIENTS)
+    # We could also combine to, cc and bcc, but it's useless for now
+
+    for to_chunk in to_chunks:
+        copy_kvs = {k: email_message.__dict__[k] for k in _EMAIL_KEYS}
+        copy_email = EmailMessage(**copy_kvs)
+        copy_email.to = to_chunk
+        sanitized_emails.append(copy_email)
+
+    return sanitized_emails
+
+
+# Custom async email backend wrapper
+# ----------------------------------
+
+# Settings are explicit for humans, but this is what Huey needs
+_NB_RETRIES = int(
+    settings.SEND_EMAIL_RETRY_TOTAL_TIME_IN_SECONDS / settings.SEND_EMAIL_DELAY_BETWEEN_RETRIES_IN_SECONDS
+)
+
+
+@task(retries=_NB_RETRIES, retry_delay=settings.SEND_EMAIL_DELAY_BETWEEN_RETRIES_IN_SECONDS, context=True)
+def _async_send_message(email_id, *, task=None):
+    with transaction.atomic():
+        try:
+            email = Email.objects.select_for_update(no_key=True).get(pk=email_id)
+        except Email.DoesNotExist:
+            # Email deleted from django admin, stop trying to send it.
+            logger.warning("Not sending email_id=%d, it does not exist in the database.", email_id)
+            return
+        message = EmailMessage(
+            from_email=email.from_email,
+            reply_to=email.reply_to,
+            to=email.to,
+            cc=email.cc,
+            bcc=email.bcc,
+            subject=email.subject,
+            body=email.body_text,
+        )
+        try:
+            with get_connection(backend=settings.ASYNC_EMAIL_BACKEND) as connection:
+                connection.send_messages([message])
+        except AnymailError as e:
+            if e.response is not None:
+                try:
+                    email.esp_response = e.response.json()
+                except InvalidJSONError:
+                    logger.exception(
+                        "Received invalid response from Mailjet, email_id=%d. Payload: %s",
+                        email_id,
+                        e.response.text,
+                    )
+            else:
+                logger.exception("Could not reach Email Service Provider.")
+            success = False
+        else:
+            try:
+                email.esp_response = message.anymail_status.esp_response.json()
+            except AttributeError as e:
+                # anymail_status is None in development and default test environments.
+                if settings.ASYNC_EMAIL_BACKEND in [
+                    "django.core.mail.backends.console.EmailBackend",
+                    "django.core.mail.backends.locmem.EmailBackend",
+                ]:
+                    success = True
+                else:
+                    raise e
```

Comment:
Il me semble qu'avec la forme sans paramètre on aura juste la TB de l'erreur originelle plutôt que d'avoir aussi ce bloc dedans, qui pour le coup n'est pas très utile je pense :
```suggestion
                    raise
```

Reply:
Tout à fait :+1: